diff --git a/libmdnsd/1035.c b/libmdnsd/1035.c
index 472c280..08ff2e4 100644
--- a/libmdnsd/1035.c
+++ b/libmdnsd/1035.c
@@ -80,7 +80,7 @@ static unsigned short int _ldecomp(const unsigned char *ptr)
 static bool _label(struct message *m, const unsigned char **bufp, const unsigned char *bufEnd, char **namep)
 {
 	int x;
-	const unsigned char *label;
+	const unsigned char *label = NULL;
 	char *name;
 
 	/* Set namep to the end of the block */
@@ -436,7 +436,6 @@ bool message_parse(struct message *m, unsigned char *packet, size_t packetLen)
 {
 	int i;
 	const unsigned char *buf;
-	m->_bufEnd = packet + packetLen;
 
 	/* Message format: https://tools.ietf.org/html/rfc1035
 
@@ -455,6 +454,7 @@ bool message_parse(struct message *m, unsigned char *packet, size_t packetLen)
 
 	if (packet == 0 || m == 0)
 		return false;
+	m->_bufEnd = packet + packetLen;
 
 	/* See https://tools.ietf.org/html/rfc1035
 	                                1  1  1  1  1  1
diff --git a/libmdnsd/mdnsd.c b/libmdnsd/mdnsd.c
index 227bb76..a4b96f3 100644
--- a/libmdnsd/mdnsd.c
+++ b/libmdnsd/mdnsd.c
@@ -740,7 +740,7 @@ int mdnsd_in(mdns_daemon_t *d, struct message *m, struct sockaddr *ip, unsigned
 			}
 
 			/* Send the matching unicast reply */
-			if (!hasConflict && port != 5353)
+			if (!hasConflict && port != 5353 /* htons(5353) */)
 				_u_push(d, r_start, m->id, ip, port);
 		}
 
@@ -750,7 +750,7 @@ int mdnsd_in(mdns_daemon_t *d, struct message *m, struct sockaddr *ip, unsigned
 	/* Process each answer, check for a conflict, and cache */
 	for (i = 0; i < m->ancount; i++) {
 		if (m->an[i].name == NULL) {
-			MDNSD_LOG_ERROR("Got answer with NULL name at %p. Type: %d, TTL: %ld\n", (void*)&m->an[i], m->an[i].type, m->an[i].ttl);
+			MDNSD_LOG_ERROR("Got answer with NULL name at %p. Type: %d, TTL: %lu\n", (void*)&m->an[i], m->an[i].type, m->an[i].ttl);
 			return 3;
 		}
 
@@ -792,7 +792,7 @@ int mdnsd_out(mdns_daemon_t *d, struct message *m, struct sockaddr *ip, unsigned
 			memset(&addr, 0, sizeof(addr));
 			addr.sin_family = AF_INET;
 			addr.sin_port = *port;
-			inet_pton(AF_INET, "224.0.0.251", &(addr.sin_addr));
+			inet_pton(AF_INET, "224.0.0.251" /* "0.0.0.0" */, &(addr.sin_addr));
 			memcpy(ip, &addr, sizeof(addr));
 	}
 	m->header.qr = 1;
@@ -806,7 +806,6 @@ int mdnsd_out(mdns_daemon_t *d, struct message *m, struct sockaddr *ip, unsigned
 
 		d->uanswers = u->next;
 		*port = u->port;
-		ip = u->to;
 		m->id = (unsigned short int)u->id;
 		message_qd(m, u->r->rr.name, u->r->rr.type, (unsigned short int)d->clazz);
 		message_an(m, u->r->rr.name, u->r->rr.type, (unsigned short int)d->clazz, u->r->rr.ttl);
